src/components/checkbox/checkbox.test.jsx

```diff
@@ -0,0 +1,233 @@
+      render(<Checkbox {...requiredProps} checked />);
+      expect(screen.getByRole('checkbox', { name })).toBeChecked();
+    });
+
+    it('updates the value', () => {
+      const { rerender } = render(<Checkbox {...requiredProps} checked={false} />);
+      expect(screen.getByRole('checkbox', { name })).not.toBeChecked();
+      rerender(<Checkbox {...requiredProps} checked={true} />);
+      expect(screen.getByRole('checkbox', { name })).toBeChecked();
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      const { rerender } = render(<Checkbox {...requiredProps} ref={ref} checked={false} />);
+      expect(ref.current.checked).toBe(false);
+
+      userEvent.click(screen.getByRole('checkbox', { name }));
+
+      rerender(<Checkbox {...requiredProps} ref={ref} checked={true} />);
+      expect(ref.current.checked).toBe(true);
+    });
+  });
+
+  describe('value api', () => {
+    it('prioritizes the value provided', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} value="i-am-good-bruh" ref={ref} />);
+      expect(ref.current.value).toBe('i-am-good-bruh');
+    });
+
+    it('sets the value to on if checked', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} checked ref={ref} />);
+      expect(ref.current.value).toBe('on');
+    });
+
+    it('sets the value to off if not checked', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} checked={false} ref={ref} />);
+      expect(ref.current.value).toBe('off');
+    });
+  });
+
+  describe('dirty api', () => {
+    it('is unset without any changes', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} ref={ref} />);
+      expect(ref.current.dirty).toBe(false);
+    });
+
+    it('is set with any changes', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} ref={ref} />);
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(ref.current.dirty).toBe(true);
+    });
+
+    it('is unset with changes back to initial state', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} ref={ref} checked />);
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(ref.current.dirty).toBe(true);
+
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(ref.current.dirty).toBe(false);
+    });
+
+    it('is unset with changes and new value prop', () => {
+      const ref = createRef();
+      const { rerender } = render(
+        <Checkbox {...requiredProps} ref={ref} checked={true} />,
+      );
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(ref.current.dirty).toBe(true);
+
+      rerender(<Checkbox {...requiredProps} ref={ref} checked={false} />);
+      expect(ref.current.dirty).toBe(false);
+    });
+  });
+
+  describe('required api', () => {
+    it('can be set', () => {
```

Comment:
Do we want to add an assertion for the `(Required)` text as well?

Reply:
f92589672bb4e96a5965b09e62c8ac1eae4c2a8b